### PR TITLE
Update types for ltx

### PR DESCRIPTION
**Pull request by BendingBender**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/xmppjs/ltx
- [x] Increase the version number in the header if appropriate.
- [x] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/ltx/index.d.ts [hunk 1]
```diff
@@ -1,17 +1,17 @@
...
-export {isNode, isElement, isText} from './lib/is';
-export {clone} from './lib/clone';
-export {createElement} from './lib/createElement';
-export {escapeXML, unescapeXML, escapeXMLText, unescapeXMLText} from './lib/escape';
-export {Parser} from './lib/Parser';
-export {parse} from './lib/parse';
-export {tag} from './lib/tag';
-export {tagString} from './lib/tagString';
-export {stringify} from './lib/stringify';
+export * from './lib/Element';
```
Comment (amcasey) [inline types/ltx/index.d.ts]: Why re-export more symbols?

File: types/ltx/lib/Element.d.ts [hunk 2]
```diff
@@ -1,21 +1,26 @@
...
  *
  * Attributes are in the element.attrs object. Children is a list of
  * either other Elements or Strings for text content.
- **/
-export declare class Element {
+ */
+export class Element {
     name: string;
-    parent: Element;
+    parent: Element | null;
```
Comment (amcasey) [inline types/ltx/lib/Element.d.ts]: Intentionally `null` here but `undefined` below?

File: types/ltx/index.d.ts (hunk 1 above)
Comment (BendingBender) [reply types/ltx/index.d.ts]: It is actually quite useful to have all other types defined in Element, like ode and TextNode. I currently add types for all of @xmpp/* libs and these exports were already quite useful there.

File: types/ltx/lib/Element.d.ts (hunk 2 above)
Comment (amcasey) [reply types/ltx/lib/Element.d.ts]: My experience has been that a lot of code bases standardize on either `null` or `undefined` and don't mix them.  I just wanted to confirm that using both was intentional.  It sounds like it is.

**Pull request merged**